### PR TITLE
MAINT: Post-release deprecations and version bumps

### PR DESCRIPTION
**Pull request by larsoner**

### CODE REVIEW

File: tools/dev/gen_css_for_mne.py [hunk 1]
```diff
@@ -16,11 +16,12 @@
 # Copyright the MNE-Python contributors.
```
Comment (larsoner) [inline tools/dev/gen_css_for_mne.py:16]: Snuck in a renaming of this file -- no reason to have it in `MNE` itself as it's really a packaging tool. This helps with stuff like automated dependency resolution as well (inspired by the autotick bot's analysis on conda-forge)

**Comment by larsoner** [comment]
@drammock ready for review/merge from my end

**Review by drammock — CHANGES_REQUESTED** [review]
looks like the version bumps in README and in GH Actions setup script aren't reflected in `pyproject.toml`?  Otherwise LGTM

**Comment by larsoner** [comment]
I realized I forgot to look for stuff/workarounds we can drop now that versions are bumped!

**Comment by larsoner** [comment]
Okay got a couple hundred more lines removed by getting rid of a bunch of `matplotlib` helpers that were `< 3.6` specific (and a few numpy/scipy etc. ones). Hopefully it comes back green, if it does I'll merge :crossed_fingers: !

**Comment by drammock** [comment]
this should also get a changelog entry in the Dependencies section too, right?

**Comment by larsoner** [comment]
@drammock okay to merge when green given the new changelog entry?

**Review by drammock — APPROVED** [review]

**Pull request merged**